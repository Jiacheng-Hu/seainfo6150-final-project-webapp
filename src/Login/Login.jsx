import React from "react";

    const Login = () => {
    return (
        <div>
            <div>Username</div>
            <input type="text" name="username" id="username"/>
            <div>Password</div>
            <input type="password"/><br/>
            <input type="submit" value="submit"/>
            <input type="reset" value="reset"/>
        </div>
    );
    
};

export default Login;