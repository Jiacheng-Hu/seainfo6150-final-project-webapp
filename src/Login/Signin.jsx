import React from "react";

const Signin = () => {
    
    return (
        <div>
            <div>Username</div>
            <input type="text" name="username" id="username"/>
            <div>First Name</div>
            <input type="text"/>
            <div>Last Name</div>
            <input type="text"/>
            <div>ID Number</div>
            <input type="text"/>
            <div>Email</div>
            <input type="text"/>
            <div>Degree</div>
            <input type="radio" name="degree"/>Bachelor
            <input type="radio" name="degree"/>Master
            <input type="radio" name="degree"/>Doctor
            <div>Major</div>
            <select>
                <option value="is">Information Systems</option>
                <option value="cs">Computer Science</option>
                <option value="data">Data Analytics Engineering</option>
                <option value="ba">Business Analytics</option>    
            </select>
            <div>Password</div>
            <input type="password"/>
            <div>Re-enter Password</div>
            <input type="password"/><br/>
            <input type="submit" value="Submit"/><br/>
            <input type="reset" value="Reset"/>        
        </div>
    );
};

export default Signin;