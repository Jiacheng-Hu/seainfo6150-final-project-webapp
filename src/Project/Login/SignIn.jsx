import React from "react";

    const SignIn = () => {
    return (
        <div className="signin">
            <h2 className="text-center">Please sign in</h2>
            <div className="margin-top">Username</div>
            <input type="text" name="username" id="username" required className="form-control"/>
            <div className="margin-top">Password</div>
            <input type="password" name="password" id="password" required  className="form-control"/><br/>
            <input type="submit" value="submit" className="btn btn-lg btn-primary btn-block margin-top"/>
            <input type="reset" value="reset" className="btn btn-lg btn-default btn-block margin-top"/>
          
        </div>
    );
    
};

export default SignIn;