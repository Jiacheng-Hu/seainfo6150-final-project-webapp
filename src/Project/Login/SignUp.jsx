import React from "react";

const SignUp = () => {
    
    return (
        <div className="signup">

            <p>Please fill in this form to sign up, fields with * is required</p>
            <div>*Username</div>
            <input type="text" name="username" id="username" required className="form-control"/>
            <div>*First Name</div>
            <input type="text" required className="form-control"/>
            <div>*Last Name</div>
            <input type="text" required className="form-control"/>
            <div>*ID Number</div>
            <input type="text" required className="form-control"/>
            <div>*Email</div>
            <input type="text" pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$" title="Must follow the valid email format" required className="form-control"/>
            <div>Degree</div>
            
            <input type="radio" name="degree" />Bachelor
            <input type="radio" name="degree"/>Master
            <input type="radio" name="degree"/>Doctor
            <div>Major</div>
            <select  className="form-control">
                <option value="is">Information Systems</option>
                <option value="cs">Computer Science</option>
                <option value="data">Data Analytics Engineering</option>
                <option value="ba">Business Analytics</option>    
            </select>
            <div>Password</div>
            <input type="password" name="password" id="password" required className="form-control"/>
            <div>Re-enter Password</div>
            <input type="password" name="password" id="password" required className="form-control"/><br/>

            <input type="submit" value="Submit" className="btn btn-lg btn-primary btn-block margin-top"/>
            <input type="reset" value="Reset" className="btn btn-lg btn-default btn-block margin-top"/>        
        </div>
    );
};

export default SignUp;