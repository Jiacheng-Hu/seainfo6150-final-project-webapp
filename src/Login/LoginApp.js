import React, {useState} from "react";
import Login from "./Login";
import Signin from "./Signin";

const LoginApp = () => {
    const [display, setDisplay] = useState("login");
    function onSubmit(e){
        e.preventDefault();
        let content = document.getElementById("username").value;
        alert(`Welcome ${content}`);
        window.location.href="/homepage";
    }
    
    return (
        <div> 
            <form onSubmit={onSubmit}>
                {display === "login" ? <Login/> : <Signin/>}
            </form>
            <button onClick={() => setDisplay("login")}>Sign-In</button>
            <button onClick={() => setDisplay("signin")}>Register</button>   
        </div>
        
    );
};

export default LoginApp;