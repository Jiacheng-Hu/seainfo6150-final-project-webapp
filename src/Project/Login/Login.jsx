import React, {useState} from "react";
import SignIn from "./SignIn";
import SignUp from "./SignUp";
import {useHistory} from "react-router-dom";

const Login = () => {
    let history = useHistory();
    const [display, setDisplay] = useState("Sign in");
    function onSubmit(e){
        e.preventDefault();
        let content = document.getElementById("username").value;
        alert(`Welcome ${content}`);
        let username=e.target.elements.username.value;
        history.push(
            "/",
            {
                username: username,
                login: true
            }
        )
    }
    
    return (
        <div> 
            <form onSubmit={onSubmit}>
                {display === "Sign in" ? <SignIn/> : <SignUp/>}
            </form>
            <button onClick={() => setDisplay("Sign in")} class="signBtn">Sign in</button>
            <button onClick={() => setDisplay("sign up")} class="signBtn">Sign up</button>   
        </div>
        
    );
};

export default Login;