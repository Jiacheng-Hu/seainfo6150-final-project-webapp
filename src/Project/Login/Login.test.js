import React from "react";
import Login from "./Login";


describe("Login tests", () => {
    it("renders correctly", () => {
        const {container} = render(<Login/>);
        expect(container).toMatchSnapshot();
    });
});