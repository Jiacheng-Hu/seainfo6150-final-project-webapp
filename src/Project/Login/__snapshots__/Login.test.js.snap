// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests renders correctly 1`] = `
<div>
  <div>
    <form>
      <div
        class="signin"
      >
        <h2
          class="text-center"
        >
          Please sign in
        </h2>
        <div
          class="margin-top"
        >
          Username
        </div>
        <input
          class="form-control"
          id="username"
          name="username"
          required=""
          type="text"
        />
        <div
          class="margin-top"
        >
          Password
        </div>
        <input
          class="form-control"
          id="password"
          name="password"
          required=""
          type="password"
        />
        <br />
        <input
          class="btn btn-lg btn-primary btn-block margin-top"
          type="submit"
          value="submit"
        />
        <input
          class="btn btn-lg btn-default btn-block margin-top"
          type="reset"
          value="reset"
        />
      </div>
    </form>
    <button
      class="signBtn"
    >
      Sign in
    </button>
    <button
      class="signBtn"
    >
      Sign up
    </button>
  </div>
</div>
`;
